sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorConnection.java

```diff
@@ -350,7 +351,11 @@ private synchronized Connection getOrCreateConnection() throws IOException {
             connection = reactor.connectionToHost(handler.getHostname(), handler.getProtocolPort(), handler);
 
             reactorExceptionHandler = new ReactorExceptionHandler();
-            executor = new ReactorExecutor(reactor, Schedulers.single(), connectionId,
+            // Use a new single-threaded scheduler for this connection as QPID's Reactor is not thread-safe.
```

Comment:
Is there anything we could do to make calls into QPID's Reactor thread safe? If so I'm wondering if it would be safer to use a thread-pooled Scheduler so that we don't create and tear down many, many threads.

Reply:
I think it is the qpid library that's not thread-safe. So, I don't think we can fix it in this library.